src/Aspire.Hosting/Keycloak/KeycloakContainerResource.cs

```diff
@@ -0,0 +1,29 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace Aspire.Hosting.ApplicationModel;
+
+/// <summary>
+/// A resource that represents a Keycloak container.
+/// </summary>
+/// <param name="name">The name of the resource.</param>
+public class KeycloakContainerResource(string name) : ContainerResource(name), IResourceWithEnvironment, IResourceWithServiceDiscovery
+{
+
+    /// <summary>
+    /// Gets the endpoint string for the Keycloak server.
+    /// </summary>
+    /// <returns>A endpoint string for the Keycloak server in the form "host:port".</returns>
+    public string? GetEndpoints()
```

Comment:
Who calls this?

Reply:
It comes from the interface IResourceWithServiceDiscovery which herited from IResourceWithBindings

Reply:
Where?

https://github.com/dotnet/aspire/blob/60b2529dedede70430db881f76c4a50eeafaf9ce/src/Aspire.Hosting/ApplicationModel/IResourceWithServiceDiscovery.cs#L11

Reply:
Sorry, I confused it with the GetEndpoint method with the argument and the GetConnectionString method from IResourceWithConnectionString.

I will remove this method.